lib/streamlit/watcher/event_based_path_watcher.py

```diff
@@ -60,12 +60,24 @@ def close_all() -> None:
         path_watcher.close()
         LOGGER.debug("Watcher closed")
 
-    def __init__(self, path: str, on_changed: Callable[[str], None]) -> None:
+    def __init__(
+        self,
+        path: str,
+        on_changed: Callable[[str], None],
+        *,  # keyword-only arguments:
+        glob_pattern: Optional[str] = None,
+        allow_nonexistent: bool = False,
```

Comment:
Nit - could you add a docstring for the constructor's params? path and on_changed are obvious, but `glob_pattern` and `allow_nonexistent` are less so.

Reply:
Ah, yeah this would be worth adding. I think I actually initially removed the constructor docstring because the types made the information pretty redundant, but with the new args being added it'd be useful again